### DISCUSSION TITLE
Manual Table Update  or table refresh

### DISCUSSION THREAD

**Discussion by MilvioLua** [dotnetcore/BootstrapBlazor#3934] [Q&A] [answered]
the code below auto-refreshes the table automatically, based on a interval
how to manually force a reload in my table?

<Table TItem="Foo"
       IsPagination="true" PageItemsSource="@(new int[] { 2, 4 })"
       IsStriped="true" IsBordered="true" IsMultipleSelect="true"
       IsAutoRefresh="@IsAutoRefresh" AutoRefreshInterval="2000"
       OnQueryAsync="@OnManualQueryAsync">
    <TableColumns>
        <TableColumn @bind-Field="@context.DateTime" Width="180" />
        <TableColumn @bind-Field="@context.Name" Width="100" />
        <TableColumn @bind-Field="@context.Address" />
        <TableColumn @bind-Field="@context.Count" />
        <TableColumn @bind-Field="@context.Complete" />
    </TableColumns>
</Table>

i would like from my code 
call something like 

 private async Task TableReload(Table table). // something that references a table
        {
        // some code
        table.Reload() or table.Refresh(); // something similar
         
        }

### Describe the solution you'd like

i would like from my code 
call something like 
 private async Task TableReload(Table table)
        {
        // some code
        table.Reload() or table.Refresh(); // something similar
         
        }

**Comment by bb-auto** [comment]
@MilvioLua Thank you for reporting. We will give triage later.

**Comment by j4587698** [comment, marked as answer]
You can ref table and call table.QueryAsync();

**Comment by MilvioLua** [comment]
I try doing this .
Markup
` <Table @ref="Table" TItem="Department" OnQueryAsync="QueryTable">`
 code
 private Table<Department> Table { get; set; }
 
 private async Task<bool> PostCreate() {
  //some code
 await Table.OnQueryAsync();
}
 
 
can you be a little bit more clear?
I'm doing this what you said Table.OnQueryAsync(); but says error: others no argument given that correspond to the required parameter

**Comment by ArgoZhang** [comment]
@MilvioLua Hi could you give this project a star? thanks.

```
await Table.QueryAsync();
```

not `Table.OnQueryAsync`

**Reply by MilvioLua** [reply]
sure! I will, but by the way, awesome job.
